active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,220 @@
+    PlaceholderText="0 (123) 456-7890"
+    FormatType="InternationalTelephone" />
+```
+
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for custom input" 
+    Header="Enter equation:" 
+    PlaceholderText="00 ft 00 in"
+    CustomFormatString="## ft ## in"
+    StripLeadingZeros="False" />
+```
+
+### Add increment and decrement stepping
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for items in order" 
+    Header="Order:" 
+    Text="2"
+    StepSize="2"
+    UpDownArrowsEnabled="True"
+    UpDownDragEnabled="True" />
+```
+
+### Add a prefix and suffix 
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for IP address" 
+    Header="Enter equation:" 
+    PlaceholderText="000.000.000.000" 
+    Prefix="IP Address:" />
+```
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for cents" 
+    Header="Enter equation:" 
+    PlaceholderText="0.00" 
+    Suffix="cents" />
+```
+
+### Add mininum and maximum values
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for items in order" 
+    Header="Order:" 
+    PlaceholderText="1" 
+    MinValue="1"
+    MaxValue="99" />
+```
+
+## Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+## API Notes
+<!-- Option 1: Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. For properties, specify the default value of the property if it
+isn't the type's default (for example an int-typed property that doesn't default to zero.) -->
+
+<!-- Option 2: Put these descriptions in the below API Details section,
+with a "///" comment above the member or type. -->
+
+## API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+```c++ 
+enum NumberBoxFormatType
+{
+    IPAddress,
+    InternationalTelephone,
+    Currency,
+};
+
+unsealed runtimeclass NumberBox : Windows.UI.Xaml.Controls.TextBox
+{
+    NumberBox();
+    
+    String Prefix;
+    String Suffix;
+    
+    Boolean AcceptsCalculation;
+    Boolean UpDownArrowsEnabled;
+    Boolean UpDownDragEnabled;
+  
+    NumberBoxFormatType FormatType;
+    String CustomFormatString;
+    Boolean StripLeadingZeros;
+   
+    Integer MinValue;
+    Integer MaxValue;
+
+    static Windows.UI.Xaml.DependencyProperty PrefixProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty SuffixProperty{ get; };
+    
+    static Windows.UI.Xaml.DependencyProperty AcceptsCalculationProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty UpDownArrowsEnabledProperty{ get; };
```

Comment:
How will the Up and Down arrows by styled, and will there be any way to customize the styling?
A full DataTemplate might be overkill but having the ability to specify button colors is a necessity.

Reply:
How much will the arrow buttons increment or decrement the value?
Surely this needs to be configurable, so we need a `StepSize` property too?

Reply:
Yes, the StepSize property has been added. As far as customization goes, great call out. Ill dig into this after we have landed on a design as we are not sure where the Up/Down buttons will go (e.g., in the control vs outside of it).

Reply:
I have added an open question for SpinButton styling.

Reply:
@SavoySchuler About the SpinButton terminology: Perhaps because I'm a non-native English speaker, but just looking at that name, I have absolutely no clue what is meant with a *SpinButton* for this control. It also isn't intuitive to me because I only know the word *spin* in a context such as "wheel/disk spinning", however, what is spinning when looking at the NumberBox control?

Reply:
@Felix-Dev Thank you for mentioning this! I really appreciate the unique perspective you can bring. 

Our [Fabric UI system names this same control SpinButton](https://developer.microsoft.com/en-us/fabric#/controls/web/spinbutton) which is what I based the name off of.  Historically, names like [Spin Button](https://www.c-sharpcorner.com/uploadfile/mahesh/spin-button-in-wpf/), [Spin Control](https://docs.microsoft.com/en-us/windows/desktop/uxguide/ctrl-spin-controls), and [Spinner ](https://docs.microsoft.com/en-us/windows/desktop/windowsribbon/windowsribbon-controls-spinner)have also been used by our older frameworks as well. 

I had it named UpDownButton briefly but re-aligned for consistency across our modern and legacy platforms. Do you have some alternate suggestions that I could consider? Otherwise, perhaps I can be proactive with clear documentation examples + images for this component?

Reply:
@SavoySchuler Sorry for the late reply! I have yet to think about a good alternative name.

Out of interest, what was the reason why this terminology was chosen in the first place? Perhaps that could help me relate the term "Spin Button" with the up/down buttons of the NumberBox control.

Adding images to the final documentation on docs.microsoft.com describing the different components of the UI sounds good to me! Perhaps similar to the [Navigation View UI documentation](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/topnav-anatomy.png) (obviously scaled down for the NumberBox control)?

Reply:
@MikeHillberg do you have any history on the origin of the "Spin Button" name? To paraphrase the [Spin Controls documentation](https://docs.microsoft.com/en-us/windows/win32/uxguide/ctrl-spin-controls), my best guess is that the analogy arises from being able to increase uni-directional change while remaining stationary, as with some of these spinning objects that increase measured value per interaction from a (semi) stationary force:

[Merry-Go-Round](http://media.treehugger.com/assets/images/2013/10/merry-go-round-power.jpg.662x0_q70_crop-scale.jpg)
[Fidget Spinner](https://timedotcom.files.wordpress.com/2017/05/170503-fidget-spinner-school-ban.jpg?quality=85)

Otherwise, yes, I can be sure the visual documentation helps to make this clear. 😄

Reply:
@mrlacey, to answer your first posted question which I appear to have overlooked (apologies there) the following APIs have been added for styling the Spin Buttons:

    Windows.UI.Xaml.Style DecrementButtonStyle;
    Windows.UI.Xaml.Style IncrementButtonStyle;